### ISSUE TITLE
Add SWITCH Statement

### ISSUE THREAD

**Issue by krisdestruction** [KSP-KOS/KOS#860]
Hi,

I was wondering if the `switch` statement could be added for flow control just like the [switch in Java/Matlab](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/switch.html). This would utilize the `break` that is already defined and use `cases` for each case.

**Comment by Dunbaratu** [comment]
There's an awful lot of language stuff I'd love to add.  But other things kept coming up first.
switch is one of them, but I wonder about the "fall through" logic of the case statements in a language meant to be easy for newbies.  that fall-through can confuse people.

**Comment by krisdestruction** [comment]
Actually I feel like it might be easy for newbies as it not only eliminates their need to make many if statements (eg. a condition to check if they are at specific altitudes), but I think newbies understand `case` better than `if` if they're new to programming. "In the case it's X do this, in the case it's Y do that". For myself, I just want to execute something at several altitudes and I see this as a major codesaver for advanced users as well, something that each language should strive to achieve anyways. Assuming you implement the standard optimizations, you might even get better performance compared to using standard `if` statements.

**Comment by Dunbaratu** [comment]
I was referring to the need to explicitly break because it falls to the next case if you don't.

**Comment by krisdestruction** [comment]
@Dunbaratu In that case, automatically break like in the [Matlab implementation](http://www.mathworks.com/help/matlab/ref/switch.html) of switch. If you don't know, Matlab was designed to be simple to use as well (not to the extent of kOS, but more than others).

**Comment by krisdestruction** [comment]
Better yet, you can do an `ezswitch` for beginners to automatically break and a `switch` for the standard implementation. `ezswitch` inheriting the functionality of `switch` for minimal coding.

**Comment by erendrake** [comment]
if we implement a switch statement (other than the switch we currently have :P) i would only want one. that one being one that doesnt allow you to fall through multiple cases. If you look at many other interpreted languages you see the common knowledge is to avoid falling through as a maintenance issue. JSLint for example shows the practice as an infraction.

**Comment by Dunbaratu** [comment]
Yes, and the proper fix would have been to invert the meaning (break if you say nothing, and require an explicit statement to fall through), rather than to punish good programmers and take away a useful feature because OTHER people can't handle it.

**Comment by krisdestruction** [comment]
I can see some users cringing, but I'm fine with either implementation. Perhaps a `continue` statement to do the fall-through?

**Comment by Zyt3x** [comment]
What about both case and switch? "case" has no fall-through, "switch" does

**Comment by krisdestruction** [comment]
`case` is typically a keyword pertaining to a `switch`. This is the same case as the 2 switches that @erendrake didn't want.
